src/libraries/System.Diagnostics.DiagnosticSource/tests/ActivityTests.cs

```diff
@@ -1508,6 +1508,54 @@ public void TestInsertingFirstTag(string key, object value, bool add, int result
             Assert.Equal(resultCount, a.TagObjects.Count());
         }
 
+        [Fact]
+        public void StructEnumerator_TagsLinkedList()
+        {
+            /*
+             * This test verifies the presence of the struct Enumerator on TagsLinkedList used by customers dynamically to avoid allocations.
+             */
+            Activity a = new Activity("TestActivity");
+            a.AddTag("Tag1", true);
+
+            IEnumerable<KeyValuePair<string, object>> enumerable = a.TagObjects;
+
+            bool foundGetEnumerator = false;
+            foreach (MethodInfo method in enumerable.GetType().GetMethods(BindingFlags.Public | BindingFlags.Instance | BindingFlags.NonPublic))
+            {
```

Comment:
you don't need foreach here. yu can request the method name directly.

Reply:
Updated